demos/testing/test_database_backed_server/README.md

```diff
@@ -0,0 +1,30 @@
+# Testing Tornado Database Backed Application.
```

Comment:
I'm not sure this is interesting enough as it is to warrant a separate example. You're just configuring an extra library in `create_app`. The tests are exactly the same as the other cases (with my suggestions below).

Some important questions about testing with databases are left out of this example. When are tables created? How do I use a different database for testing vs production? But the answers to those questions aren't really specific to testing *with tornado*. (except maybe the fact that you need to use `run_sync` in `setUp` to create your tables)

Another pattern we should encourage if this example is going to get more complicated is to move as much logic as possible out of your handler classes so it can be tested directly without going through `self.fetch()` (which makes it much harder to debug and see what's going on).